system/BaseModel.php

```diff
@@ -343,6 +345,15 @@ protected function initialize()
     {
     }
 
+    /**
+     * Provides a shared instance of the Query Builder.
+     *
+     * @return BaseBuilder
+     *
+     * @throws ModelException
+     */
+    abstract public function builder(?string $table = null);
```

Comment:
Why does BaseModel have `builder()`?
Any kind of database should have the Query Builder?

Reply:
Ya, trying to understand the difference between BaseModel and Model. It seems Model is based on Builder but BaseModel is not.